mobs/nakki_boss/FightDirector.cs

```diff
@@ -99,30 +90,10 @@ private void OnLilypadAttackCompleted(int attackId) {
 		}
 	}
 
-	private void DelayedLilypadAttack(LilypadAttackStats stats) {
-		var timer = new Timer() {
-			Autostart = true,
-			OneShot = true,
-			WaitTime = stats.Delay,
-		};
-
-		_delayedAttacks.Add(timer);
-
-		timer.Timeout += () => {
-			_delayedAttacks.Remove(timer);
-			timer.QueueFree();
-			LilypadArena.SinkLilypads(stats);
-		};
-		AddChild(timer);
-	}
-
 	private void Reset() {
-		foreach (var timer in _delayedAttacks) {
-			timer.Stop();
-			timer.QueueFree();
-		}
-		_delayedAttacks.Clear();
-		_attacksWaitingForNakkiSignal.Clear();
+		_lilypadAttackCancelationTokenSource.Cancel();
+		_lilypadAttackCancelationTokenSource = new();
```

Comment:
This _should_ mark cancellation tokens of any existing lilypad attack tasks as cancelled, causing them to throw `OperationCancelledException` next time they check the token.

New source is then immediately initialized to allow executing new tasks with non-cancelled tokens.